include/pacemaker.h

```diff
@@ -0,0 +1,137 @@
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#ifndef PACEMAKER__H
+#  define PACEMAKER__H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \file
+ * \brief High Level API
+ * \ingroup pacemaker
+ */
+
+#  include <crm/stonith-ng.h>
+#  include <libxml/tree.h>
+
+/*!
+ * \brief Perform a STONITH action.
+ *
+ * \param[in] st        A connection to the STONITH API.
+ * \param[in] target    The target of the given action.
+ * \param[in] action    The action to perform.
+ * \param[in] name
+ * \param[in] timeout   How long to wait for the operation to complete (in seconds).
+ * \param[in] tolerance
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_action(stonith_t *st, const char *target, const char *action,
+                        const char *name, int timeout, int tolerance);
+
+/*!
+ * \brief List the fencing operations that have occurred for a specific node.
+ *
+ * \param[in,out] xml       The destination for the result, as an XML tree.
+ * \param[in]     st        A connection to the STONITH API.
+ * \param[in]     st_opts   Bitfield of stonith_call_options.
+ * \param[in]     target    The node to get history for.
+ * \param[in]     timeout   How long to wait for the operation to complete (in seconds).
+ * \param[in]     quiet     Suppress most output.
+ * \param[in]     verbose   Include additional output.
+ * \param[in]     broadcast Gather fencing history from all nodes.
+ * \param[in]     cleanup   Clean up fencing history after listing.
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_history(xmlNodePtr *xml, stonith_t *st, unsigned int st_opts,
+                         char *target, int timeout, bool quiet, int verbose,
+                         bool broadcast, bool cleanup);
+
+/*!
+ * \brief List all installed STONITH agents.
+ *
+ * \note If the xml parameter is not NULL, it will be freed first and the previous
+ *       contents lost.
+ *
+ * \param[in,out] xml     The destination for the result, as an XML tree.
+ * \param[in]     st      A connection to the STONITH API.
+ * \param[in]     timeout How long to wait for the operation to complete (in seconds).
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_installed(xmlNodePtr *xml, stonith_t *st, int timeout);
+
+/*!
+ * \brief When was a device last fenced?
+ *
+ * \note If the xml parameter is not NULL, it will be freed first and the previous
+ *       contents lost.
+ *
+ * \param[in,out] xml       The destination for the result, as an XML tree.
+ * \param[in]     target    The target of the fencing operation.
+ * \param[in]     as_nodeid 
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_last_fenced(xmlNodePtr *xml, const char *target, bool as_nodeid);
+
+int pcmk_stonith_list_targets(xmlNodePtr *xml, stonith_t *st, unsigned int st_opts,
+                              char *target, int timeout);
```

Comment:
I'm a little confused by the uses of target, device, name, and agent throughout these files I've made.  This would be a nice time to straighten all that out too.

Reply:
target = node being fenced
agent = a fence agent executable
device = stonith-class resource configured in the CIB (agent + parameters)
name = arbitrary label for logging (e.g. "node1 successfully fenced for <name>")